group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -671,6 +688,10 @@ private void throwIfConsumerGroupHeartbeatRequestIsInvalid(
             if (request.subscribedTopicNames() == null || request.subscribedTopicNames().isEmpty()) {
                 throw new InvalidRequestException("SubscribedTopicNames must be set in first request.");
             }
+        } else if (request.memberEpoch() == LEAVE_GROUP_STATIC_MEMBER_EPOCH) {
+            throwIfEmptyString(request.memberId(), "MemberId can't be empty. GroupId: " + request.groupId());
+            throwIfNull(request.instanceId(), "InstanceId can't be null for Static Member. GroupId: "
```

Comment:
I think that the instance id cannot be null and cannot be empty as well. Then let's use `InstanceId can't be null or empty.`